config/v1/types_oauth.go

```diff
@@ -1,37 +1,518 @@
+
+	// Provider-specific configuration
+	// The json tag MUST match the `Type` specified above, case-insensitively
+	// e.g. For `Type: "LDAP"`, the `LDAPPasswordIdentityProvider` configuration should be provided
+
+	// BasicAuth contains configuration options for the BasicAuth IdP
+	// +optional
+	BasicAuth *BasicAuthPasswordIdentityProvider `json:"basicAuth,omitempty"`
+
+	// AllowAll enables the AllowAllIdentityProvider which provides identities for users
+	// authenticating using non-empty passwords.
+	// Defaults to `false`, i.e. allowAll set to off
+	// +optional
+	AllowAll bool `json:"allowAll,omitempty"`
+
+	// DenyAll enables the DenyAllPasswordIdentityProvider which provides no identities for users
+	// Defaults to `false`, ie. denyAll set to off
+	// +optional
+	DenyAll bool `json:"denyAll,omitempty"`
+
+	// HTPasswd enables user authentication using an HTPasswd file to validate credentials
+	// +optional
+	HTPasswd *HTPasswdPasswordIdentityProvider `json:"htpasswd,omitempty"`
+
+	// LDAP enables user authentication using LDAP credentials
+	// +optional
+	LDAP *LDAPPasswordIdentityProvider `json:"ldap,omitempty"`
+
+	// Keystone enables user authentication using keystone password credentials
+	// +optional
+	Keystone *KeystonePasswordIdentityProvider `json:"keystone,omitempty"`
+
+	// RequestHeader enables user authentication using request header credentials
+	RequestHeader *RequestHeaderIdentityProvider `json:"requestHeader,omitempty"`
+
+	// GitHub enables  user authentication using GitHub credentials
+	// +optional
+	GitHub *GitHubIdentityProvider `json:"github,omitempty"`
+
+	// GitLab enables user authentication using GitLab credentials
+	// +optional
+	GitLab *GitLabIdentityProvider `json:"gitlab,omitempty"`
+
+	// Google enables user authentication using Google credentials
+	// +optional
+	Google *GoogleIdentityProvider `json:"google,omitempty"`
+
+	// OpenID enables user authentication using OpenID credentials
+	// +optional
+	OpenID *OpenIDIdentityProvider `json:"openID,omitempty"`
+}
+
+// BasicAuthPasswordIdentityProvider provides identities for users authenticating using HTTP basic auth credentials
+type BasicAuthPasswordIdentityProvider struct {
+	// RemoteConnectionInfo contains information about how to connect to the external basic auth server
+	OAuthRemoteConnectionInfo `json:",inline"`
+}
+
+// RemoteConnectionInfo holds information necessary for establishing a remote connection
+type OAuthRemoteConnectionInfo struct {
+	// URL is the remote URL to connect to
+	URL string `json:"url"`
+	// CA is a reference to a ConfigMap containing the CA for verifying TLS connections
+	CA ConfigMapReference `json:"ca"`
+
+	// TLSClientCert references a secret containing the TLS client certificate to present when
+	// connecting to the server.
+	// Looks under the key "tls.cert" for the data unless a lookup key is specified in the secret ref
+	TLSClientCert LocalSecretReference `json:"tlsClientCert"`
```

Comment:
optional followup.  I think you can reasonably accept just the name and require the key since tls.key and tls.crt are standard.